src/liballoc/collections/btree/node.rs

```diff
@@ -571,19 +579,13 @@ impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
     }
 
     fn into_key_slice_mut(mut self) -> &'a mut [K] {
-        // Same as for `into_key_slice` above, we try to avoid a run-time check.
-        if (mem::align_of::<NodeHeader<K, V, K>>() > mem::align_of::<NodeHeader<K, V>>()
-            || mem::size_of::<NodeHeader<K, V, K>>() != mem::size_of::<NodeHeader<K, V>>())
-            && self.is_shared_root()
-        {
-            &mut []
-        } else {
-            unsafe {
-                slice::from_raw_parts_mut(
-                    MaybeUninit::first_ptr_mut(&mut (*self.as_leaf_mut()).keys),
-                    self.len(),
-                )
-            }
+        debug_assert!(!self.is_shared_root());
+        // We cannot be the shared root, so `as_leaf_mut` is okay.
```

Comment:
We? I'd prefer something more objective like the slice or whatever it is

Reply:
I didn't actually write that comment, and think it's somewhat superfluous, but please come up with a concrete suggestion here, like:

- // `self` cannot be the shared root
- // This `NodeRef` cannot be the shared root
- // Our node reference cannot be the shared root

Reply:
I would go with: 
>  self cannot be the shared root

Reply:
I don't think that's any better, on the contrary, it comes across as poor grammar and capitalization to me. Also, there are already at least 3 comments of this "we = self" style in the file, so at least one author and one reviewer didn't object.

Reply:
The style of the public API documentation of Option, Vector, BTreeSet, leads to (in unscientifically determined order of popularity):
- // The node cannot be the shared root
- // The `NodeRef` cannot be the shared root
If there are multiple instances involved, we explicitly use `self` (with backticks). Somehow, its awkward capitalization is carefully avoided, except [Vec::split_off](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_off)